src/components/DropdownSelect/DropdownSelect.tsx

```diff
@@ -0,0 +1,55 @@
+import { Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+
+type DropdownSelectProps = {
+    options: Array<string>,
+    variant: SelectVariant.single | SelectVariant.checkbox
+    selectedProp: string | Array<string>,
+    setSelected: Dispatch<SetStateAction<any>>, // TODO type this
```

Comment:
Your type will likely look something like this: (type: 'Name/URL' | 'Version' | 'Architecture') => void;

Reply:
I'm not sure this works. The issue here is I wanted to be able to set a string (or filter in this case) or Array<string> (for versions/arches). The "single" variant would be setting a string, the "checkbox" variant would be setting an array.

Reply:
Then just add both as the types with an "|" or operator?

Reply:
It doesn't seem to be as simple as that. I pushed up what I tried, but I can't seem to find a way to type this that doesn't complain.

`setSelected: Dispatch<SetStateAction<Array<string>>> | Dispatch<SetStateAction<Filters>>`

Reply:
setSelected: Dispatch<SetStateAction<Filters | Array<string>>>;

Reply:
After further attempts at fixing this, I am now onboard for the "any" method within the DropdownSelect. 😅